x-pack/plugin/kql/src/main/java/org/elasticsearch/xpack/kql/KqlPlugin.java

```diff
@@ -7,10 +7,21 @@
 
 package org.elasticsearch.xpack.kql;
 
+import org.elasticsearch.Build;
 import org.elasticsearch.plugins.ExtensiblePlugin;
 import org.elasticsearch.plugins.Plugin;
 import org.elasticsearch.plugins.SearchPlugin;
+import org.elasticsearch.xpack.kql.query.KqlQueryBuilder;
+
+import java.util.List;
 
 public class KqlPlugin extends Plugin implements SearchPlugin, ExtensiblePlugin {
+    @Override
+    public List<QuerySpec<?>> getQueries() {
+        if (Build.current().isSnapshot()) {
```

Comment:
Should we check the `SearchCapabilities` here instead?

Reply:
It is not perfect like this but I do not think I will change it because it would mean changing the visibility of 

```
private static final String KQL_QUERY_SUPPORTED = "kql_query";
```

or using a hardcoded value:

```
if (SearchCapabilities.capabilities.contains("kql_query")) {
```